pkg/detectors/youneedabudget/youneedabudget.go

```diff
@@ -20,14 +21,15 @@ var _ detectors.Detector = (*Scanner)(nil)
 var (
 	client = common.SaneHttpClient()
 
-	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
-	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"youneedabudget"}) + `\b([0-9a-f]{64})\b`)
+	// YNAB Personal Access Tokens are 64-character hexadecimal strings
+	// Based on documentation: access tokens are used with Bearer authentication
+	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"youneedabudget", "ynab"}) + `["']?([A-Za-z0-9_-]{43,44})["']?`)
```

Comment:
Are we sure the old pattern keys are no longer valid? Also, we don’t need `["']` in the regex - let’s just capture the exact pattern.